### ISSUE TITLE
LibWeb: Crash when using `<link rel="icon">` inside an iframe with content set using innerHTML

### ISSUE THREAD

**Issue by utf-4096** [SerenityOS/serenity#24106]
Creating a `<link rel="icon">` tag inside an iframe using `innerHTML` causes the following crash:
```
VERIFICATION FAILED: m_about_base_url.has_value() at /opt/serenity/Userland/Libraries/LibWeb/DOM/Document.cpp:953
```

```
#0  ak_verification_failed () at /opt/serenity/AK/Assertions.cpp:108
#1  0x00007f36e704f844 in fallback_base_url () at /opt/serenity/Userland/Libraries/LibWeb/DOM/Document.cpp:953
#2  base_url () at /opt/serenity/Userland/Libraries/LibWeb/DOM/Document.cpp:973
#3  parse_url () at /opt/serenity/Userland/Libraries/LibWeb/DOM/Document.cpp:983
#4  0x00007f36e71a1955 in inserted () at /opt/serenity/Userland/Libraries/LibWeb/HTML/HTMLLinkElement.cpp:81
#5  0x00007f36e70a6d7d in operator() () at /opt/serenity/Userland/Libraries/LibWeb/DOM/Node.cpp:480
#6  for_each_shadow_including_inclusive_descendant<(lambda at /opt/serenity/Userland/Libraries/LibWeb/DOM/Node.cpp:478:72)>(void) ()
    at /opt/serenity/Meta/Lagom/../../Userland/Libraries/LibWeb/DOM/ShadowRoot.h:76
#7  0x00007f36e70a6471 in insert_before () at /opt/serenity/Userland/Libraries/LibWeb/DOM/Node.cpp:478
#8  0x00007f36e7214ea3 in insert_foreign_element () at /opt/serenity/Userland/Libraries/LibWeb/HTML/Parser/HTMLParser.cpp:776
#9  0x00007f36e720a943 in insert_html_element () at /opt/serenity/Userland/Libraries/LibWeb/HTML/Parser/HTMLParser.cpp:793
#10 0x00007f36e720a943 in handle_in_head () from /opt/serenity/Build/lagom/lib64/liblagom-web.so.0
#11 0x00007f36e720a6c0 in process_using_the_rules_for () at /opt/serenity/Userland/Libraries/LibWeb/HTML/Parser/HTMLParser.cpp:382
#12 handle_before_head () at /opt/serenity/Userland/Libraries/LibWeb/HTML/Parser/HTMLParser.cpp:836
#13 0x00007f36e7207e6e in run () at /opt/serenity/Userland/Libraries/LibWeb/HTML/Parser/HTMLParser.cpp:202
#14 0x00007f36e72091af in run () at /opt/serenity/Userland/Libraries/LibWeb/HTML/Parser/HTMLParser.cpp:223
#15 0x00007f36e721d24b in parse_html_fragment () at /opt/serenity/Userland/Libraries/LibWeb/HTML/Parser/HTMLParser.cpp:4249
#16 0x00007f36e70c30c7 in parse_fragment () at /opt/serenity/Userland/Libraries/LibWeb/DOMParsing/InnerHTML.cpp:23
#17 0x00007f36e70c334a in inner_html_setter () at /opt/serenity/Userland/Libraries/LibWeb/DOMParsing/InnerHTML.cpp:42
#18 0x00007f36e7078464 in set_inner_html () at /opt/serenity/Userland/Libraries/LibWeb/DOM/Element.cpp:713
#19 0x00007f36e75eac48 in operator() () at /opt/serenity/Build/lagom/Userland/Libraries/LibWeb/Bindings/ElementPrototype.cpp:2596
#20 throw_dom_exception_if_needed<(lambda at /opt/serenity/Build/lagom/Userland/Libraries/LibWeb/Bindings/ElementPrototype.cpp:2596:62), Web::WebIDL::ExceptionOr<void>, JS::Value> ()
    at /opt/serenity/Meta/Lagom/../../Userland/Libraries/LibWeb/Bindings/ExceptionOrUtils.h:97
#21 inner_html_setter () at /opt/serenity/Build/lagom/Userland/Libraries/LibWeb/Bindings/ElementPrototype.cpp:2596
#22 0x000055e6d77f115d in AK::Function<JS::ThrowCompletionOr<JS::Value> (JS::VM&)>::CallableWrapper<JS::ThrowCompletionOr<JS::Value> (*)(JS::VM&)>::call(JS::VM&) ()
    at /opt/serenity/Meta/Lagom/../../AK/Function.h:192
#23 0x00007f36e602078f in operator() () at /opt/serenity/Meta/Lagom/../../AK/Function.h:125
#24 0x00007f36e60205fa in call () at /opt/serenity/Userland/Libraries/LibJS/Runtime/NativeFunction.cpp:239
#25 0x00007f36e602008a in internal_call () at /opt/serenity/Userland/Libraries/LibJS/Runtime/NativeFunction.cpp:167
...
```

Reduced repro:
```html
<iframe id="ifr"></iframe>
<script>
    ifr.onload = () => {
        ifr.contentDocument.documentElement.innerHTML = '<link rel="icon">';
    }

    ifr.srcdoc = '';
</script>
```

Curiously, this does not seem to happen when the link element is created using `createElement`, or using `srcdoc` directly:
```html
<iframe id="ifr"></iframe>
<script>
    ifr.onload = () => {
        const elm = document.createElement('link');
        elm.setAttribute('rel', 'icon');
        ifr.contentDocument.head.appendChild(elm);
    }

    ifr.srcdoc = '';
</script>
```

**Comment by ADKaster** [comment]
Does this repro with the patch from https://github.com/SerenityOS/serenity/pull/24101 applied?

**Comment by utf-4096** [comment]
Just tested, I'm unable to repro with this patch.

**Comment by awesomekling** [comment]
Fixed by #24101
